### PR TITLE
type assertion in WrappedServeHTTP fixed

### PR DESCRIPTION
**Pull request by dencoded**
added fix for https://github.com/TykTechnologies/tyk/issues/1339

Unfortunately, I wasn't able to write good unit test because `WSDialer` is doing hijack for connection, so it is not trivial - need to invest more time to research the best approach to test. I was thinking about `ResponseRecorder` with `http.Hijacker` implemented.

### CODE REVIEW

File: reverse_proxy.go [hunk 1]
```diff
@@ -518,12 +518,18 @@ func (p *ReverseProxy) WrappedServeHTTP(rw http.ResponseWriter, req *http.Reques
 	// Circuit breaker
 	breakerEnforced, breakerConf := p.CheckCircuitBreakerEnforced(p.TykAPISpec, req)
 
+	// set up TLS certificates for upstream if needed
+	var tlsCertificates []tls.Certificate = nil
```
Comment (buger) [inline reverse_proxy.go]: Do you added `= nil` for expressivity or it is needed somehow?

**Comment by buger** [comment]
It will be perfect you can add simple test covering websockets, because right now we have nothing.

File: reverse_proxy.go (hunk 1 above)
Comment (dencoded) [reply reverse_proxy.go]: it is not needed as zero value is already nil, I think I added it just looking at " = nil" assignment in  https://github.com/TykTechnologies/tyk/pull/1342/files#diff-da06648202ab00313675c5150a92bffeL524

**Comment by dencoded** [comment]
ok, I will add some simple test for reverse proxy to websocket

**Comment by dencoded** [comment]
@buger I've added test for reverse proxying to websocket upstream as requested, please take a look

**Pull request merged**